internal/wasm/store.go

```diff
@@ -362,8 +363,13 @@ func (s *Store) instantiate(
 		return nil, err
 	}
 
+	var allocator experimental.MemoryAllocator
+	if ctx != nil {
+		allocator, _ = ctx.Value(ctxkey.MemoryAllocatorKey{}).(experimental.MemoryAllocator)
+	}
```

Comment:
We don't check ctx != nil elsewhere, so please simplify as 
```suggestion
	allocator := ctx.Value(ctxkey.MemoryAllocatorKey{}).(experimental.MemoryAllocator)
```

Reply:
Actually, there's a specific test case covering this:
https://github.com/tetratelabs/wazero/blob/f31be308fa39150af4939fec0e1f0bb005cd11f4/internal/wasm/module_instance_test.go#L87-L89

Reply:
nil contexts should not be common in Go. There's even a lint to use context.Background() instead of passing nil because a lot of places assume a valid context. It may be reasonable to change this test.